packages/blocks/src/api/utils.js

```diff
@@ -39,21 +38,16 @@ const ICON_COLORS = [ '#191e23', '#f8f9f9' ];
  * @return {boolean} Whether the block is an unmodified block.
  */
 export function isUnmodifiedBlock( block ) {
-	// Cache a created default block if no cache exists or the default block
-	// name changed.
-	if ( ! isUnmodifiedBlock[ block.name ] ) {
-		isUnmodifiedBlock[ block.name ] = createBlock( block.name );
-	}
-
-	const newBlock = isUnmodifiedBlock[ block.name ];
-	const blockType = getBlockType( block.name );
-
-	function isEqual( a, b ) {
-		return ( a?.valueOf() ?? a ) === ( b?.valueOf() ?? b );
-	}
+	return Object.entries( getBlockType( block.name )?.attributes ?? {} ).every(
+		( [ key, definition ] ) => {
+			// Every attribute that has a default must match the default.
+			if ( definition.hasOwnProperty( 'default' ) ) {
+				return block.attributes[ key ] === definition.default;
+			}
 
-	return Object.keys( blockType?.attributes ?? {} ).every( ( key ) =>
-		isEqual( newBlock.attributes[ key ], block.attributes[ key ] )
+			// Every attribute that does not have a default must be empty.
+			return ! block.attributes[ key ]?.length;
```

Comment:
If we want to be a bit more strict here, we could also check if the type is rich text. If not, the value must be undefined. If yes, the value's length must be 0.

Rich text is a bit different from the rest because it has an implicit default that is an empty `RichTextData` instance.